CMakeLists.txt

```diff
@@ -24,6 +30,25 @@ option(BUILD_DOXYGEN "Enable/Disable the generation of the Doxygen." OFF)
 option(BUILD_EXAMPLES "Enable/Disable the compilation of the examples." OFF)
 option(BUILD_TESTS "Enable/Disable the compilation of unit tests." ON)
 option(BUILD_TOOLS "Enable/Disable the compilation of tools." ON)
+option(BUILD_SHARED_LIBS "Enable/disable build of shared libraries." ON)
+option(BUILD_SHARED_EXTENSIONS "Enable/disable build of shared extensions." ON)
+option(BUILD_STATIC_LIBS "Enable/disable build of static libraries." ON)
+option(BUILD_STATIC_EXTENSIONS "Enable/disable build of static extensions." ON)
```

Comment:
It's arguable, but I think we should not allow people to mix static and dynamic libraries. I propose to keep only `BUILD_SHARED_LIBS` and `BUILD_STATIC_LIBS`. I wonder if we should add an option for each extension library